base/associative.jl

```diff
@@ -64,6 +64,8 @@ in(k, v::KeyIterator) = get(v.dict, k, secret_table_token) !== secret_table_toke
     keys(a::Associative)
 
 Return an iterator over all keys in a collection.
+`keys(Dict)` and `values(Dict)` both iterate the `Dict`
+ and return the elements in the same order.
 `collect(keys(d))` returns an array of keys.
 Since the keys are stored internally in a hash table,
 the order in which they are returned may vary.
```

Comment:
Since this paragraph talks about order, better add your new sentence below this one, starting with "Though" or "But"?